### DISCUSSION TITLE
How to use debugger for opam

### DISCUSSION THREAD

**Discussion by Jomosoto** [ocaml/opam#6037] [Q&A] [answered]
Hey there!

I want to investigate a section in this project with a debugger, but I haven't understood the build process enough to figure out how to do it yet. Is it possible to make a bytecode build or is there another way to use a debugger for the opam code?

Thanks!

**Comment by kit-ty-kate** [comment, marked as answer]
it is possible to make a bytecode build but you'll need to add `(modes (byte exe))` in this stanza: https://github.com/ocaml/opam/blob/62b041a4dc65eb599d4b61a2404769c6adfbcd25/src/client/dune#L14

I've done it once but i can't remember it if was the only thing to do. Be aware though that the resulting binary is going to be extremely slow (from what i remember)

**Reply by Jomosoto** [reply]
Thanks! That worked. For anyone in the future getting errors: You havo to run a `make clean` before.
